src/OpenLoco/src/Vehicles/VehicleHead.cpp

```diff
@@ -179,12 +182,158 @@ namespace OpenLoco::Vehicles
         calculateRefundCost();
     }
 
+    // 0x004B8340
+    static void RecalculateTrainMinReliability(VehicleHead& head)
+    {
+        Vehicle train(head);
+        const auto getReliabilty = [](const Car& car) -> uint16_t {
+            if (car.front->reliability == 0)
+            {
+                return 0xFFFFU;
+            }
+            return car.front->reliability;
+        };
+        const auto minReliability = (*std::min_element(train.cars.begin(), train.cars.end(), [&getReliabilty](const Car& carLhs, const Car& carRhs) {
+                                        return getReliabilty(carLhs) < getReliabilty(carRhs);
+                                    })).front->reliability;
+        train.veh2->reliability = minReliability == 0xFFFFU ? 0 : minReliability / 256;
+    }
+
     // 0x004B9509
     void VehicleHead::updateDaily()
     {
-        registers regs{};
-        regs.esi = X86Pointer(this);
-        call(0x004B9509, regs);
+        bool resetStoppedTimeout = true;
+        if (status == Status::stopped)
+        {
+            if (mode == TransportMode::road)
+            {
+                if (tileX != -1
+                    && hasVehicleFlags(VehicleFlags::commandStop))
+                {
+                    if (Tutorial::state() == Tutorial::State::none)
+                    {
```

Comment:
Could we simplify these `if` blocks a little to reduce nesting? All `else` branches just set `resetStoppedTimeout = false`, after all.

Reply:
I'm not sure how to do that and keep the same results. Not everything has an else block as they don't all reset the stopped timeout. If it was as simple as all have an else then yes i could see how to do it.

Reply:
It seems like it is a bit better to use `resetStoppedTimeout = false` as initial value both logically (by default we don't need to do anything) and for readability (we don't need 1 else in this case):
```c++
bool resetStoppedTimeout = false;
if (status == Status::stopped)
{
    if (mode == TransportMode::road)
    {
        if (tileX != -1
            && hasVehicleFlags(VehicleFlags::commandStop))
        {
            if (Tutorial::state() == Tutorial::State::none)
            {
                var_79++;
                if (var_79 >= 20)
                {
                    GameCommands::VehicleChangeRunningModeArgs args{};
                    args.head = head;
                    args.mode = GameCommands::VehicleChangeRunningModeArgs::Mode::startVehicle;
                    auto regs = static_cast<Interop::registers>(args);
                    regs.bl = GameCommands::Flags::apply;
                    GameCommands::vehicleChangeRunningMode(regs);
                    resetStoppedTimeout = true;
                }
            }
        }
        else
        {
            resetStoppedTimeout = true;
        }
    }
}
else
{
    resetStoppedTimeout = true;
}
```
[The change process](https://gist.github.com/Lev135/cc06a189ced5ce19b3362dae9a30c48c) was very simple, so I hope I don't broke anything. But, please, check me too

Reply:
You can trvially combine the inner-most `if` if you inline the `var_79++`:

```
 if (status == Status::stopped)
 {
     if (mode == TransportMode::road)
     {
         if (tileX != -1 && hasVehicleFlags(VehicleFlags::commandStop))
         {
             if (Tutorial::state() == Tutorial::State::none && ++var_79 >= 20)
             {
                 GameCommands::VehicleChangeRunningModeArgs args{};
                 args.head = head;
                 args.mode = GameCommands::VehicleChangeRunningModeArgs::Mode::startVehicle;
                 auto regs = static_cast<Interop::registers>(args);
                 regs.bl = GameCommands::Flags::apply;
                 GameCommands::vehicleChangeRunningMode(regs);
             }
             else
             {
                 resetStoppedTimeout = false;
             }
         }
     }
     else
     {
         resetStoppedTimeout = false;
     }
 }
```